htlcswitch/interceptable_switch.go

```diff
@@ -617,6 +635,20 @@ func (f *interceptedForward) Resume() error {
 	return f.htlcSwitch.ForwardPackets(nil, f.packet)
 }
 
+// ResumeModified resumes the default behavior with field modifications.
+func (f *interceptedForward) ResumeModified(
+	outgoingAmountMsat lnwire.MilliSatoshi,
+	customRecords record.CustomSet) error {
+
+	// Modify the htlc.
+	f.packet.amount = outgoingAmountMsat
```

Comment:
So we if take a look at the packet struct: https://github.com/lightningnetwork/lnd/blob/1c229e44227e090fc21bb3ab32a74ddf7e7e5d5f/htlcswitch/packet.go#L11-L110

What we want to modify here is the `htlc` field itself. _This_ is the outgoing HTLC that the incoming link has already prepared to send to outgoing link.

Reply:
The `customRecords` field here is actually the [field from the _onion_](https://github.com/lightningnetwork/lnd/blob/1c229e44227e090fc21bb3ab32a74ddf7e7e5d5f/htlcswitch/link.go#L3469) that we're passing along. We started to pass this along so the interceptor could introspect into the onion custom records of the incoming HTLC.